rego/rego.go

```diff
@@ -299,6 +299,13 @@ func (r *Rego) eval(ctx context.Context, compiled ast.Body, txn storage.Transact
 		t.Input = r.input
 	}
 
+	t.Cancel = topdown.NewCancel()
+
```

Comment:
here: complete := make(chan struct{}); defer close(complete), and then pass the channel for waitForDone, so it doesn't hang forever (in case of context not cancelled)